.github/workflows/build-test.yaml

```diff
@@ -92,7 +92,63 @@ jobs:
       env:
         testsuite: ${{ matrix.tests-type }}
       run: python -m nose -c nose_${testsuite}.cfg --traverse-namespace
-    - name: Report Failed Answers
-      if: ${{ matrix.tests-type == 'answer' && failure() }}
+
+
+  image-tests:
+    name: Image tests
+    runs-on: ubuntu-latest
+
+    concurrency:
+      # auto-cancel any in-progress job *on the same branch*
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.11'
+
+    - name: Checkout repo (with submodules)
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Install dependencies and yt
       shell: bash
-      run: python tests/report_failed_answers.py -f -m --xunit-file answer_nosetests.xml
+      env:
+        dependencies: ''
+      run: |
+        source ./tests/ci_install.sh
+        python -m pip install cartopy scipy
+
+    - name: Run Image Tests
+      run: |
+        pytest --color=yes --mpl -m mpl_image_compare \
+               --mpl-generate-summary=html \
+               --mpl-results-path=pytest_mpl_results \
+               --mpl-baseline-path=tests/pytest_mpl_baseline
+
+    - name: Generate new image baseline
```

Comment:
Just so I understand, this will only run in case of failure, and in that particular case it will upload the artifacts?

Reply:
Not exactly. Note that steps that do the uploads are conditioned with `always()`, so we try to upload as many artefacts as we produced. The "new baseline" artifact may not exist because it's only generated when needed, but the html report should always be present.